.github/workflows/changelog-generation.yml

```diff
@@ -0,0 +1,38 @@
+name: Generate Changelog
+on:
+  push:
+    branches: master
+jobs:
+  generate-changelog:
+    name: Run automation script
+    runs-on: ubuntu-latest
+    steps:
+    - name: Check out script code
+      uses: actions/checkout@v3
+      with:
+        repository: ciscoecosystem/release_script
+        path: ./release_script
+
+    - name: Check out collection code
+      uses: actions/checkout@v3
+      with:
+        path: ./collection
+
+    - name: Set up Python 3.8
+      uses: actions/setup-python@v1
+      with:
+        python-version: 3.8
+
+    - name: Install release script requirements
+      run: pip install -r ./release_script/requirements.txt
+
+    - name: "Get github token"
+      run: echo "$GITHUB_TOKEN" > ./release_script/token.txt
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+    - name: "GH auth"
+      run: gh auth login --with-token < ./release_script/token.txt
+
+    - name: Run automation script
+      run: python ./release_script/script.py
```

Comment:
Have you try directly setting the environment variable for the python script like this:
```
- name: Run automation script
   run: python ./release_script/script.py
   env:
      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
This would replace all 3 last steps